kingpin/actors/rightscale/api.py

```diff
@@ -247,18 +246,21 @@ def find_by_name_and_keys(self, collection, name, exact=True, **kwargs):
             RightScale Resource Objects
         """
         filter_keys = []
-        kwargs['name'] = name
         for key, val in kwargs.items():
             filter_keys.append('%s==%s' % (key, val))
-        params = {'filter[]': filter_keys}
+        params = {'filter[]': sorted(filter_keys)}
 
         found = collection.index(params=params)
         if not exact and len(found) > 0:
             return found
 
-        for f in found:
-            if f.soul['name'] == name:
-                return f
+        if len(found) < 1:
+            return None
+
+        if len(found) == 1:
```

Comment:
This is pretty bad... There shouldn't be any guesswork at to what _type_ of data gets returned. None, Object, or List. I strongly recommend always returning a list. List of 0, list of 1 or list of many...